R/contribution_matrix.R

```diff
@@ -724,8 +724,12 @@ CreateContributionMatrix <- function(data, treatment_ids, outcome_type, outcome_
                                                treatments = treatments,
                                                outcome_type = outcome_type,
                                                outcome_measure = outcome_measure)$effect_sizes$Effect
-    #Multiply the n-th row of 'contribution' by the n-th element of 'effect_sizes'
-    contribution <- contribution * effect_sizes
+    #Create an effect size matrix in order to do the multiplication in the next step
+    effect_size_matrix <- matrix(rep(effect_sizes, each = nrow(contribution)),
+                                 nrow = nrow(contribution)
+                                 )
+    #Multiply the n-th column of 'contribution' by the n-th element of 'effect_sizes'
+    contribution <- contribution * effect_size_matrix
```

Comment:
Whilst your method does work and is only one extra line, I think you can keep it to 1 line rather than 2 with
```
contribution <- t(t(contribution)*effect_sizes)
```

Reply:
I tried transposing `effect_sizes` but it didn't work. I did not think about transposing `contribution`!

Reply:
Haha, I mostly thought of it from the conversation we had about which way round the multiplication should be, and you said it's because of the transposition :D